src/_terminal-utils.ts

```diff
@@ -3,6 +3,11 @@ import process from 'node:process';
 import * as ansiEscapes from 'ansi-escapes';
 import stringWidth from 'string-width';
 
+// Synchronized output mode escape sequences (DEC mode 2026)
+// These prevent tearing during updates by buffering terminal operations
+const SYNC_START = '\x1B[?2026h';
+const SYNC_END = '\x1B[?2026l';
```

Comment:
don't you find this escape characters in `ansi-escapes` ?

Reply:
At least I need to know where these code comes from. if you find a resource, add it as a comment.

Reply:
it's a non-standard thing, though a lot of terminals implement it.
<https://gist.github.com/christianparpart/d8a62cc1ab659194337d73e399004036>
I tested removing it from the PR and it looks like the rest of the changes make far more impact than this one alone.

Should I just remove it?

Reply:
added that link

Reply:
Thank you. you can add the escape character if you have a proper reason! 
https://gist.github.com/christianparpart/d8a62cc1ab659194337d73e399004036
I think this doc is kind a reasonable. you can add it as a comment.